File: arrays/C_Cpp/shuffle.c
#include <stdlib.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* shuffle(int* nums, int numsSize, int n, int* returnSize){
    int* arr = malloc(numsSize * sizeof(int));
    *returnSize = numsSize;
    int x = 0;
    int y = n;
    int i = 0;
    while (i < numsSize) {
        if (i % 2 == 0 && x < n) {
            arr[i] = nums[x];
            x++;
            i++;
        }
        else if (i % 2 == 1 && y < numsSize) {
            arr[i] = nums[y];
            y++;
            i++;
        }
    }
    return arr;
}

File: arrays/python/shuffle.py

def shuffle(nums: list[int], n:  int) -> int:
    arr = []
    x = 0
    y = n
    for i in range(len(nums)):
        if i % 2 == 0:
            arr.append(nums[x])
            x += 1
        else:
            arr.append(nums[y])
            y += 1
    return arr
